packages/server/src/ee/controllers/v1/boards/filter.ts

```diff
@@ -15,49 +19,226 @@ import {
+    }
+
+    let totalCount: number | null = null;
+    let totalPages: number | null = null;
+    let currentPage = 1;
+    let hasNextPage = false;
+
+    if (!page) {
+      const boardsMetaData = await getBoardMetaData({ after });
+
+      if (boardsMetaData) {
+        totalCount = boardsMetaData.totalBoardsCount;
+        totalPages = Math.ceil(boardsMetaData.totalBoardsCount / first);
+        hasNextPage = boardsMetaData.remainingBoardsCount - first > 0;
+      }
+
+      if (after) {
+        const seenBoards = totalCount - boardsMetaData.remainingBoardsCount;
+        currentPage = Math.floor(seenBoards / first) + 1;
+      }
+    }
+
+    res.status(200).send({
+      status: {
+        code: 200,
+        type: "success",
+      },
+      boards,
+      results: boards,
+      ...(page
+        ? {}
+        : {
+            page_info: {
+              count: boardsLength,
+              current_page: currentPage,
+              has_next_page: hasNextPage,
+              end_cursor: endCursor,
+              start_cursor: startCursor,
+            },
+            total_pages: totalPages,
+            total_count: totalCount,
+          }),
+    });
+  } catch (err) {
+    logger.log({
+      level: "error",
+      message: err,
+    });
+
+    res.status(500).send({
+      message: error.general.serverError,
+      code: "SERVER_ERROR",
+    });
+  }
+}
+
+export async function getBoards({
+  first,
+  after,
+  created,
+  page,
+}: IGetBoardQueryOptions) {
+  try {
+    let boardsQuery = database<IBoardPrivate>("boards")
       .select(
         "boards.boardId",
         "boards.name",
         "boards.color",
         "boards.url",
         "boards.createdAt",
+        "display",
+        "view_voters",
       )
       .count("posts", { as: "post_count" })
-      .from("boards")
       .leftJoin("posts", "boards.boardId", "posts.boardId")
       .where({
-        display: true,
+        "boards.display": true,
       })
       .groupBy("boards.boardId")
       .orderBy("boards.createdAt", created)
-      .limit(limit)
-      .offset(limit * (page - 1));
+      .orderBy("boards.boardId", created)
+      .limit(first);
 
-    res.status(200).send({ boards });
-  } catch (err) {
+    if (page) {
+      boardsQuery = boardsQuery.offset(first * (page - 1));
+    } else if (after) {
+      const afterBoard = await database("boards")
+        .select("createdAt", "boardId")
+        .where("boardId", after)
+        .first();
+
+      if (!afterBoard) return [];
+
+      const operator = created === "ASC" ? ">" : "<";
+
+      boardsQuery = boardsQuery.where(function () {
+        this.where("boards.createdAt", operator, afterBoard.createdAt).orWhere(
+          function () {
+            this.where("boards.createdAt", "=", afterBoard.createdAt).andWhere(
+              "boards.boardId",
+              operator,
+              after,
+            );
+          },
+        );
```

Comment:
@TaranaKhanna can you elaborate on the usage of the `OR` clauses here for the boards query?